### PR TITLE
Additional fixes to PR 777

### PR DESCRIPTION
**Pull request by Neeratyoy**
#### Reference Issue
PR #777. 

#### What does this PR implement/fix? Explain your changes.
Makes the changes/improvements suggested by @PGijsbers that were not addressed prior to merge.

### CODE REVIEW

**Comment by PGijsbers** [comment]
Checks manually cancelled.

**Comment by Neeratyoy** [comment]
@PGijsbers wrt to why an experimental `HistGradientBoostingClassifier` is used in [this example](https://github.com/openml/openml-python/pull/777#discussion_r510715345):

To handle mixed-type data frames with missing values, we need to create a transformation which imputes these missing values for both categorical and continuous types. Under the current OpenML setup, we cannot create a flow with two instances of the same class, a `SimpleImputer` in this case. So far, for [various](https://github.com/openml/openml-python/blob/bf3cd2ebaac10bd05809a1ce90e346248c4c61b1/tests/test_runs/test_run_functions.py#L570) [unit tests](https://github.com/openml/openml-python/blob/bf3cd2ebaac10bd05809a1ce90e346248c4c61b1/tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py#L1706), we have been handling it via an alias such as [this](https://github.com/openml/openml-python/blob/develop/openml/testing.py#L261). 

To prevent importing from `openml.testing` in a documentation example, it was decided to bypass the need of a second imputer. @mfeurer suggested the `HistGradientBoostingClassifier` since it can handle numeric missing values implicitly. Therefore, for this example, a `passthrough` works.

**Comment by PGijsbers** [comment]
That makes sense (even though it is a bit unfortunate). However, why not consider creating the example on a task which does not have two types of missing values? While all examples implicitly depend on some version(s) of scikit-learn, it feels using an experimental feature has exceptionally narrow support. Old versions (`<=0.20`) won't have this, future versions won't have this.

File: openml/extensions/sklearn/extension.py [hunk 1]
```diff
@@ -1608,23 +1610,22 @@ def _prediction_to_probabilities(y: np.ndarray, model_classes: List[Any]) -> pd.
...
-            if isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
-                if task.class_labels is not None:
-                    if isinstance(y_train, np.ndarray) and isinstance(task.class_labels[0], str):
-                        # mapping (decoding) the predictions to the categories
-                        # creating a separate copy to not change the expected pred_y type
-                        y = [task.class_labels[pred] for pred in y]
-                else:
-                    raise ValueError("The task has no class labels")
-            else:
+            if not isinstance(task, (OpenMLClassificationTask, OpenMLLearningCurveTask)):
```
Comment (PGijsbers) [inline openml/extensions/sklearn/extension.py:1613]: Still, as far as I can tell this function is only called whenever this condition is already satisfied?

File: openml/extensions/sklearn/extension.py [hunk 2]
```diff
@@ -1608,23 +1610,22 @@ def _prediction_to_probabilities(y: np.ndarray, model_classes: List[Any]) -> pd.
...
                 return None
 
-            # y: list or numpy array of predictions
+            if task.class_labels is None:
```
Comment (PGijsbers) [inline openml/extensions/sklearn/extension.py:1616]: Still, as far as I can tell this function is only called whenever this condition is already satisfied?

**Review by PGijsbers — CHANGES_REQUESTED** [review]

File: openml/extensions/sklearn/extension.py (hunk 2 above)
Comment (Neeratyoy) [reply openml/extensions/sklearn/extension.py]: I now understand. I think I can do away with the if...else under the exception block and make it cleaner?
And abstract away the checks to the `_prediction_to_probabilities` function.

**Comment by Neeratyoy** [comment]
> However, why not consider creating the example on a task which does not have two types of missing values? 

Yeah I was wondering why that was not done here but then I think we need to change the design of the example. At this point, it samples 3 tasks randomly from the suite. Do we then fix this task list? If so, do we provide a justification? 

Either way, I shall try and find 3 tasks where missing values if present is restricted to one of the types. We can then re-iterate and replace the `HistGradientBoostingClassifier` with a more stable classifier.

File: examples/30_extended/run_setup_tutorial.py [hunk 3]
```diff
@@ -58,7 +54,7 @@
 # many potential hyperparameters. Of course, the model can be as complex and as
 # easy as you want it to be
 
-from sklearn.ensemble import HistGradientBoostingClassifier
+from sklearn.ensemble import RandomForestClassifier
```
Comment (mfeurer) [inline examples/30_extended/run_setup_tutorial.py:57]: Could you please move these import up to the other imports.

File: examples/30_extended/study_tutorial.py [hunk 4]
```diff
@@ -104,7 +103,7 @@ def cat(X):
     [("cat", cat_imp, cat), ("cont", FunctionTransformer(lambda x: x, validate=False), cont)]
 )
 clf = sklearn.pipeline.Pipeline(
-    steps=[("transform", ct), ("estimator", HistGradientBoostingClassifier()),]
+    steps=[("transform", ct), ("estimator", HistGradientBoostingClassifier())]
```
Comment (mfeurer) [inline examples/30_extended/study_tutorial.py:106]: Here we actually also have the imports in two places, could you please clean these as well?

**Review by mfeurer — COMMENTED** [review]
Looks good to me. I'll approve once I see that the tests are green.

File: examples/30_extended/study_tutorial.py [hunk 5]
```diff
@@ -77,38 +72,18 @@
...
-    return X.dtypes != "category"
-
-
-def cat(X):
-    return X.dtypes == "category"
+# Model to be used
+clf = RandomForestClassifier()
 
+# We'll create a study with one run on 5 datasets present in the suite
+tasks = [115, 259, 307]
```
Comment (PGijsbers) [inline examples/30_extended/study_tutorial.py:81]: I count three?

Comment (Neeratyoy) [reply examples/30_extended/study_tutorial.py:81]: aah missed it during re-iteration
I'll push now or wait for tests to go green first?

Comment (PGijsbers) [reply examples/30_extended/study_tutorial.py:81]: I normally just push immediately.

**Review by PGijsbers — APPROVED** [review]

**Pull request merged**